### PR TITLE
Create a better menu for mobile & bonus fixes

### PR DESCRIPTION
**Pull request by badrange**
Bonus: 
* Add a workaround for Silent renew problems (just for testing today & tomorrow)
* Prevent page from overflowing horizontally on mobile
* Persist profile information in Redux

### CODE REVIEW

File: src/common/components/dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -53,21 +54,27 @@ const Dropdown: React.FunctionComponent<DropdownProps> = ({
...
-                </Button>
-              );
+              /**
+               * If the item is already displayed above
+               * the dropDown menu, we don't want to repeat it.
+               **/
+              if (!option.skipItem || index > 0)
+                return (
+                  <Button
+                    className={styles.dropdownContentOption}
```
Comment (midori0507) [inline src/common/components/dropdown/Dropdown.tsx]: Skipitem is not a quite optimal solution, just use index > 1 here

File: src/domain/app/header/userDropdown/UserDropdown.tsx [hunk 2]
```diff
@@ -1,56 +1,80 @@
 import * as React from 'react';
 import { useTranslation } from 'react-i18next';
 import { useSelector, useDispatch } from 'react-redux';
+import { useHistory } from 'react-router';
```
Comment (midori0507) [inline src/domain/app/header/userDropdown/UserDropdown.tsx]: React router dom

File: src/domain/app/header/userDropdown/UserDropdown.tsx [hunk 3]
```diff
@@ -1,56 +1,80 @@
...
+      // Clear backend auth data
+      dispatch(resetBackendAuthentication());
+      // Flush data in redux store and localStorage
+      persistor.flush();
+      logoutTunnistamo();
+    },
+  };
+
+  const user = {
+    label: (profileData && profileData.firstName) || 'User(FIXME)',
```
Comment (midori0507) [inline src/domain/app/header/userDropdown/UserDropdown.tsx]: FIXME? use a default name: User?

File: src/domain/app/layout/container.module.scss [hunk 4]
```diff
@@ -5,9 +5,11 @@
     display: grid;
 
     padding-bottom: $containerPaddingBottom;
-    @include respond-below(sm) {
-        * > div {padding: 0 $basePadding }
-    }
+    // It seemed like a good idea to add this on Friday December 13, but
+    // I'm not so sure: those paddings keep adding up for nested divs.
```
Comment (midori0507) [inline src/domain/app/layout/container.module.scss]: Just remove it, no need to keep comment code

File: src/domain/auth/authenticate.ts [hunk 5]
```diff
@@ -63,7 +63,17 @@ export const authenticateWithBackend = (
...
       type: toast.TYPE.ERROR,
     });
-    Sentry.captureException(error);
-    dispatch(fetchTokenError(error));
+    try {
+      // This is a workaround that can save us until we can fix silentRenew().
+      loginTunnistamo();
+      // eslint-disable-next-line no-console
+      console.log(
+        'Silent renew/backend token fetch fail, logging in through Tunnistamo again'
```
Comment (midori0507) [inline src/domain/auth/authenticate.ts]: No console log, warn or toast and wrap translationn

File: src/domain/app/header/navbar/Navbar.tsx [hunk 6]
```diff
@@ -15,11 +15,15 @@ const Navbar: React.FunctionComponent = props => {
     <div className={styles.navbarTop}>
       <div className={styles.logoWrapper}>
         <div className={styles.logo} onClick={() => history.push('/')}></div>
-        <h3 className={styles.appName}>{t('appName')}</h3>
+        {!isSmallScreen ? (
```
Comment (midori0507) [inline src/domain/app/header/navbar/Navbar.tsx]: !IsSmallScreen && ...
No need return ''

File: src/domain/app/header/navbar/smallScreenNav/SmallScreenNav.tsx [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+              onClick={() => toggleNavMenu(!isOpen)}
+            >
+              <Icon
+                className={styles.icon}
+                src={close}
+                alt={t('common.closeButton.altText')}
+              />
+            </button>
+          </div>
+          <UserDropdown isSmallScreen={true} />
```
Comment (midori0507) [inline src/domain/app/header/navbar/smallScreenNav/SmallScreenNav.tsx]: Remove the props if they always true, also.check.inside.component

File: src/domain/app/header/navbar/Navbar.tsx (hunk 6 above)
Comment (badrange) [reply src/domain/app/header/navbar/Navbar.tsx]: Good tip, thank you.

File: src/domain/app/header/navbar/smallScreenNav/SmallScreenNav.tsx (hunk 7 above)
Comment (badrange) [reply src/domain/app/header/navbar/smallScreenNav/SmallScreenNav.tsx]: Thanks. This is actually correct - I'll make a note to add an issue to refactor this "when we have time".

**Pull request merged**